src/braket/reservations/reservations.py

```diff
@@ -0,0 +1,82 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from contextlib import contextmanager
+
+from braket.aws.aws_device import AwsDevice
+from braket.devices.device import Device
+from braket.devices.local_simulator import LocalSimulator
+
+
+@contextmanager
+def reservation(device: Device, reservation_arn: str | None) -> Device | None:
+    """
```

Comment:
given this takes a Device, I'm not sure how to make it work with a qiskit-braket BraketAwsBackend, or even pennylane qml.device.

Reply:
This is usually an indicator that there might be an opportunity to work at a different abstraction. Would it help if we moved this functionality to a different abstraction?